core/server/data/schema/commands.js

```diff
@@ -178,12 +213,20 @@ async function addForeign({fromTable, fromColumn, toTable, toColumn, cascadeDele
  * @param {Object} configuration.transaction.knex - knex instance
  */
 async function dropForeign({fromTable, fromColumn, toTable, toColumn, transaction}) {
+    const isSQLite = db.knex.client.config.client === 'sqlite3';
+    if (isSQLite) {
+        const foreignKeyExists = await hasForeignSQLite({fromTable, fromColumn, toTable, toColumn, transaction});
+        if (!foreignKeyExists) {
+            logging.warn(`Skipped dropping foreign key from ${fromTable}.${fromColumn} to ${toTable}.${toColumn} - foreign key does not exist`);
+            return;
+        }
+    }
     try {
         logging.info(`Dropping foreign key from ${fromTable}.${fromColumn} to ${toTable}.${toColumn}`);
 
         //disable and re-enable foreign key checks on sqlite because of https://github.com/knex/knex/issues/4155
         let foreignKeysEnabled;
-        if (db.knex.client.config.client === 'sqlite3') {
+        if (isSQLite) {
             foreignKeysEnabled = await db.knex.raw('PRAGMA foreign_keys;');
```

Comment:
Hey @tpatel - are we purposefully not using the passed in transaction/connection here, and opting for using `db.knex` directly?

Reply:
Not purposefully, so this change is fine!

Reply:
I think that can be done as part of cleanup tho IMO

Reply:
I meant I think the transaction/knex instance should give the same client as db.knex.client.config.client, right?